### PR TITLE
Bugfixes addto.phtml

### PR DESCRIPTION
**Pull request by loekvangool**
Several issues: The separator is printed when not needed, code is executed when not needed, and the `<ul>` is printed when not needed.

### CODE REVIEW

File: app/design/frontend/rwd/default/template/catalog/product/view/addto.phtml [hunk 1]
```diff
@@ -21,32 +21,35 @@
 ?>
 
 <?php $_product = $this->getProduct(); ?>
-<?php $_wishlistSubmitUrl = $this->helper('wishlist')->getAddUrlWithCustomParams($_product, [], false); ?>
-<?php $_params = $this->escapeHtml(json_encode(['form_key' => $this->getFormKey()])); ?>
+<?php $_compareUrl = $this->helper('catalog/product_compare')->getAddUrlCustom($_product, false); ?>
+<?php $separator = ""; ?>
```
Comment (kiatng) [inline app/design/frontend/rwd/default/template/catalog/product/view/addto.phtml:25]:
```suggestion
<?php $_wishlistSubmitUrl = $this->helper('wishlist')->isAllow() ? $this->helper('wishlist')->getAddUrlWithCustomParams($_product, [], false) : '' ?>
<?php $_compareUrl = $this->helper('catalog/product_compare')->getAddUrlCustom($_product, false); ?>
```

File: app/design/frontend/rwd/default/template/catalog/product/view/addto.phtml [hunk 2]
```diff
@@ -21,32 +21,35 @@
...
-        </a>
-    </li>
-<?php endif ?>
-
-</ul>
+<?php if ($this->helper('wishlist')->isAllow() || $_compareUrl) : ?>
+    <ul class="add-to-links">
+    <?php if ($this->helper('wishlist')->isAllow()) : ?>
+        <?php $separator = '<span class="separator">|</span>'; ?>
+        <?php $_wishlistSubmitUrl = $this->helper('wishlist')->getAddUrlWithCustomParams($_product, [], false); ?>
```
Comment (kiatng) [inline app/design/frontend/rwd/default/template/catalog/product/view/addto.phtml:31]:
```suggestion
<?php if ($_wishlistSubmitUrl || $_compareUrl) : ?>
    <ul class="add-to-links">
    <?php if ($_wishlistSubmitUrl) : ?>
```

File: app/design/frontend/rwd/default/template/catalog/product/view/addto.phtml [hunk 3]
```diff
@@ -21,32 +21,35 @@
...
+                <?php echo $this->__('Add to Wishlist') ?>
+            </a>
+        </li>
+    <?php endif ?>
+    <?php
+    ?>
+    <?php if($_compareUrl) : ?>
+        <?php $_params = $this->escapeHtml(json_encode(['form_key' => $this->getFormKey()])) ?>
+        <li>
+            <?php echo $separator ?>
```
Comment (kiatng) [inline app/design/frontend/rwd/default/template/catalog/product/view/addto.phtml:45]:
```suggestion
    <?php if($_compareUrl) : ?>
        <?php $_params = $this->escapeHtml(json_encode(['form_key' => $this->getFormKey()])) ?>
        <li>
            <?php if ($_wishlistSubmitUrl): ?><span class="separator">|</span><?php endif ?>
```

**Review by kiatng — COMMENTED** [review]
Minor changes

File: app/design/frontend/rwd/default/template/catalog/product/view/addto.phtml [hunk 4]
```diff
@@ -21,32 +21,30 @@
...
+            <a href="#"
+               class="link-compare"
+               onclick="customFormSubmit('<?php echo $_compareUrl ?>', '<?php echo $_params ?>', 'post')">
+                <?php echo $this->__('Add to Compare') ?>
+            </a>
+        </li>
+    <?php endif ?>
 
+    </ul>
+<?php endif ?>
```
Comment (fballiano) [inline app/design/frontend/rwd/default/template/catalog/product/view/addto.phtml:50]: missing opening if

Comment (fballiano) [reply app/design/frontend/rwd/default/template/catalog/product/view/addto.phtml:50]: PHP Parse error:  syntax error, unexpected token "endif", expecting end of file in app/design/frontend/rwd/default/template/catalog/product/view/addto.phtml on line 50

File: app/design/frontend/rwd/default/template/catalog/product/view/addto.phtml [hunk 5]
```diff
@@ -21,32 +21,30 @@
...
-        <a href="#"
-           class="link-compare"
-           onclick="customFormSubmit('<?php echo $_compareUrl ?>', '<?php echo $_params ?>', 'post')">
-            <?php echo $this->__('Add to Compare') ?>
-        </a>
-    </li>
-<?php endif ?>
-
-</ul>
+    <ul class="add-to-links">
```
Comment (fballiano) [inline app/design/frontend/rwd/default/template/catalog/product/view/addto.phtml:27]:
```suggestion
<?php if($_wishlistSubmitUrl || $_compareUrl): ?>
    <ul class="add-to-links">
```

File: app/design/frontend/rwd/default/template/catalog/product/view/addto.phtml [hunk 6]
```diff
@@ -21,32 +21,31 @@
...
+        <?php $_params = $this->escapeHtml(json_encode(['form_key' => $this->getFormKey()])) ?>
+        <li>
+            <?php if ($_wishlistSubmitUrl): ?><span class="separator">|</span><?php endif ?>
+            <a href="#"
+               class="link-compare"
+               onclick="customFormSubmit('<?php echo $_compareUrl ?>', '<?php echo $_params ?>', 'post')">
+                <?php echo $this->__('Add to Compare') ?>
+            </a>
+        </li>
+    <?php endif ?>
```
Comment (fballiano) [inline app/design/frontend/rwd/default/template/catalog/product/view/addto.phtml:49]:
```suggestion
```

File: app/design/frontend/rwd/default/template/catalog/product/view/addto.phtml [hunk 7]
```diff
@@ -21,32 +21,30 @@
...
+    </ul>
 <?php endif ?>
```
Comment (fballiano) [inline app/design/frontend/rwd/default/template/catalog/product/view/addto.phtml:50]:
```suggestion
<?php if ($_wishlistSubmitUrl || $_compareUrl): ?>
    <ul class="add-to-links">
        <?php if ($_wishlistSubmitUrl) : ?>
            <li>
                <a href="<?php echo $_wishlistSubmitUrl ?>"
                   onclick="productAddToCartForm.submitLight(this, this.href); return false;"
                   class="link-wishlist">
                    <?php echo $this->__('Add to Wishlist') ?>
                </a>
            </li>
        <?php endif ?>
        <?php if ($_compareUrl) : ?>
            <?php $_params = $this->escapeHtml(json_encode(['form_key' => $this->getFormKey()])) ?>
            <li>
                <?php if ($_wishlistSubmitUrl): ?><span class="separator">|</span><?php endif ?>
                <a href="#"
                   class="link-compare"
                   onclick="customFormSubmit('<?php echo $_compareUrl ?>', '<?php echo $_params ?>', 'post')">
                    <?php echo $this->__('Add to Compare') ?>
                </a>
            </li>
        <?php endif ?>
    </ul>
<?php endif ?>
```

**Review by fballiano — APPROVED** [review]

**Review by kiatng — APPROVED** [review]
LGTM.

**Pull request merged**